spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/data/AvroDataTest.java

```diff
@@ -605,4 +723,172 @@ public void testRowLineage() throws Exception {
             record.copy(Map.of("id", 4L, "data", "d", "_row_id", 1_001L)),
             record.copy(Map.of("id", 5L, "data", "e"))));
   }
+
+  @Test
+  public void testUnshreddedVariant() throws IOException {
+    Assumptions.assumeThat(supportsVariant()).as("Variant support is not implemented").isTrue();
+
+    Schema schema =
+        new Schema(required(0, "id", LongType.get()), optional(1, "data", Types.VariantType.get()));
+
+    writeAndValidate(schema);
+  }
+
+  @Test
+  public void testShreddedVariant() throws IOException {
```

Comment:
I would probably remove this test as well. The only thing that it verifies is that `VariantReaderBuilder` was used to create the Parquet reader, but that's clear from the implementation. I think the `testTypeSchema(VariantType)` case is enough.